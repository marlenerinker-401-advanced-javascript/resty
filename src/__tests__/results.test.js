import React from 'react';

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Results from '../results.js';
const testFunction = jest.fn();

describe('Results tests', () => {
  it('should render at application start', () => {
    let app = mount(<Results headers={{}} data={[]} archive={[]} populate={testFunction}/>);
    expect(app.find('#results').exists()).toBe(true);
  });

  it('should render with data ', () => {
    let app = mount(<Results headers={{}} data={[{name: 'test'}, {name: 'test2'}]} archive={[{name: 'test'}, {name: 'test2'}]} populate={testFunction}/>);
    expect(app.find('#results').exists()).toBe(true);
    expect(app.find('#details').exists()).toBe(true);
    expect(app.find('#history').exists()).toBe(true);
  });


  it('renders correctly', () => {
    const DOM = renderer.create(<Results headers={{}} data={[{name: 'test'}, {name: 'test2'}]} archive={[{name: 'test'}, {name: 'test2'}]} populate={testFunction}/>).toJSON();
    expect(DOM).toMatchSnapshot();
  });

});
