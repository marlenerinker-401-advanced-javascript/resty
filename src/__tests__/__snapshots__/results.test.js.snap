// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results tests renders correctly 1`] = `
<div
  id="results"
>
  <section
    id="history"
  >
    <ul>
      <li
        id={0}
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<pre class=\\"__json-pretty-error__\\">undefined undefined</pre>",
            }
          }
          id="json-pretty"
          onClick={[Function]}
        />
      </li>
      <li
        id={1}
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<pre class=\\"__json-pretty-error__\\">undefined undefined</pre>",
            }
          }
          id="json-pretty"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <section
    id="details"
  >
    <ul
      id="headers"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<pre class=\\"__json-pretty__\\">{}</pre>",
          }
        }
        id="json-pretty"
      />
    </ul>
    <ul
      id="items"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<pre class=\\"__json-pretty__\\">[
  {
    \\"<span class=\\"__json-key__\\">name</span>\\": <span class=\\"__json-string__\\">\\"test\\"</span>
  },
  {
    \\"<span class=\\"__json-key__\\">name</span>\\": <span class=\\"__json-string__\\">\\"test2\\"</span>
  }
]</pre>",
          }
        }
        id="json-pretty"
      />
    </ul>
  </section>
</div>
`;
